### PR TITLE
Added main page

### LINKED ISSUE #16: Add Main Page UI (effort 4)
Finish main page view
Add /hotels/{city} page - it should show title and get all hotels for individual cases

1. initial view (you should style everything to look good - this requires Ireks Ticket to be done #4 )
image

2. after clicking City image
after clicking City, it should show all hotels in the city (we dont need to verify if hotel is available or not) - navigate to '/hotels/{city}
image

3. after searching hotels (using search bar)
we should move to page hotels/{city}
image

NOTE that there is different title in point 2 and 3

### PR DESCRIPTION
**Pull request by GoraJakub**

### CODE REVIEW

File: src/routes/noMatch.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
```
Comment (pdybowski) [inline src/routes/noMatch.js]: I will move it general routes

File: src/components/Login.jsx [hunk 2]
```diff
@@ -95,14 +96,16 @@ export class Login extends React.Component {
...
             />
-            <Button
-              size="small"
-              variant="contained"
-              color="secondary"
-              style={{ margin: '0 0 0 20px' }}
-            >
-              register
-            </Button>
+            <Link to="/register" style={{ textDecoration: 'none' }}>
```
Comment (pdybowski) [inline src/components/Login.jsx]: this is not part of the login

File: src/components/NoMatch.jsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+export const NoMatch = () => {
+  return (
+    <div>
+      <h1>404 Not Found</h1>
```
Comment (pdybowski) [inline src/components/NoMatch.jsx]: move it to the center, and change
image

File: src/components/views/mainPageView/CityCard.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+export const CityCard = ({ url, city, cityDesc }) => {
+  const classes = useStyles()
+  return (
+    <Link
+      to={{ pathname: `/hotels/${city}`, state: { city: city } }}
+      className={classes.noUnderLine}
+    >
+      <Card className={classes.root}>
+        <CardActionArea>
+          <CardMedia className={classes.media} image={url} title={city} />
```
Comment (pdybowski) [inline src/components/views/mainPageView/CityCard.jsx:46]: add some kidn of hover to it,  lets say we want to make image wider inside image box (by 2 or 3px)
image

File: src/components/views/mainPageView/Hotels.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <div>
+      {hotels.hotels && hotels.hotels.length > 0 ? (
+        data.startDate ? (
+          <h1>All free hotels in {data.city}</h1>
+        ) : (
+          <h1>All hotels in {data.city}</h1>
+        )
+      ) : (
+        <h1>No hotels found in {data.city}</h1>
```
Comment (pdybowski) [inline src/components/views/mainPageView/Hotels.jsx:36]: message should be in the center (fix it everywhere, not only about no hotels)

File: src/hooks/useSearch.js [hunk 6]
```diff
@@ -5,14 +5,23 @@ export const useSearch = () => {
...
+    if (!startDate || !endDate) {
+      data = {
+        city,
+      }
+    } else {
+      data = {
         city,
-        startDate: new Date(startDate).toISOString(),
-        endDate: new Date(endDate).toISOString(),
+        startDate,
```
Comment (pdybowski) [inline src/hooks/useSearch.js]: why???? it wont work!

File: src/hooks/useSearch.js [hunk 7]
```diff
@@ -5,14 +5,23 @@ export const useSearch = () => {
...
+        endDate,
```
Comment (pdybowski) [inline src/hooks/useSearch.js]: it wont work again!

**Review by pdybowski — CHANGES_REQUESTED** [review]
fix comments

we are waiting for hotels and we are displaying mssage that there are no hotels at the same time LOL. We should wait for the response. Add some loading bar or something...
image

File: src/hooks/useSearch.js (hunk 7 above)
Comment (GoraJakub) [reply src/hooks/useSearch.js]: I'm waiting for #30 to be approved

File: src/components/views/mainPageView/Hotels.jsx (hunk 5 above)
Comment (GoraJakub) [reply src/components/views/mainPageView/Hotels.jsx:36]: I was planning to improve visuals after #6 was done

**Comment by pdybowski** [comment]
and merge changes from main(or master) branch

File: src/hooks/useSearch.js (hunk 7 above)
Comment (pdybowski) [reply src/hooks/useSearch.js]: it is

**Pull request merged**